sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -421,6 +421,24 @@ class PlannerSuite extends SharedSparkSession {
     }
   }
 
+  test("SPARK-30036: EnsureRequirements replace Exchange " +
+      "if child has SortExec and RoundRobinPartitioning") {
```

Comment:
How about just saying `Remove unnecessary RoundRobinPartitioning` in the test title? Also, can you make the PR title obivious, too?

Reply:
Good for test title, thanks.
But it is not suitable for PR title, there are other situations in this titile.

Reply:
How about `Avoid RoundRobinPartitioning that EnsureRequirements Redundantly adds`?

Reply:
Because HashPartitioning should also be concerned.